lbrynet/conf.py

```diff
@@ -246,7 +251,7 @@ def _convert_value(value):
     'data_dir': (str, default_data_dir),
     'data_rate': (float, .0001),  # points/megabyte
     'delete_blobs_on_remove': (bool, True),
-    'dht_node_port': (int, 4444),
+    'dht_node_port' : (int, None),
     'download_directory': (str, default_download_dir),
```

Comment:
Not sure if this is the best way to do this since ADJUSTABLE_SETTINGS is meant to store the default value.. so if you do the daemon API command  settings_get, you will get back that dht_node_port and peer_port is None if user haven't set them, which can cause confusion. 

Another problem is if the user wants to set the dht_node_port to some other non default value but also still wants to UPNP to map to any port, this is not possible.

I think alternatively, we can create another setting like 'upnp_any_port_mapping' that turns this feature on/off.

Reply:
> Not sure if this is the best way to do this since ADJUSTABLE_SETTINGS is meant to store the default value.. so if you do the daemon API command settings_get, you will get back that dht_node_port and peer_port is None if user haven't set them, which can cause confusion.

I see - this wasn't clear from the code. I will fix that.

> Another problem is if the user wants to set the dht_node_port to some other non default value but also still wants to UPNP to map to any port, this is not possible.

Right, but apparently this was not wanted by design (check https://github.com/lbryio/lbry/pull/973#issuecomment-340514271).

> I think alternatively, we can create another setting like 'upnp_any_port_mapping' that turns this feature on/off. This solution seems cleaner to me.

Either strategy works for me. Maybe @lyoshenka has an opinion here?

Reply:
>> I think alternatively, we can create another setting like 'upnp_any_port_mapping' that turns this feature on/off. This solution seems cleaner to me.

> Either strategy works for me. Maybe @lyoshenka has an opinion here?

by the way, my opinion about this point: if a user is configuring a specific port, his goal is probably to let that port be the one used "outside" (i.e. due to specific firewall rules) and therefore does not want the auto-selection to be in place.

I am sure there might be some use cases where having the possibility to set a specific port *and* having the automatic external port selection enabled might be helpful.
However, I think those cases are not worth the introduction of yet another config knob because it will increase the configuration complexity for very little gain.

Reply:
I get your above immediate above point. I don't have a strong preference for either way, but if we are going to go with the current implementation, than the dht_node_port in ADJUSTABLE_SETTINGS should be set to the default value and not None (and than we turn upnp any port map on/off depending on if the port is set to default value) OR, we need to create a mechanism in the conf.py that detects if some setting has been touched by the user or not because that doesn't exist right now.  Otherwise as I mentioned before, API command settings_get will return None for the port values which is not informative.

Reply:
> OR, we need to create a mechanism in the conf.py that detects if some setting has been touched by the user or not because that doesn't exist right now.

This is probably a good the cleanest approach (setting None was just a way to work around that missing part).